unit-tests/artifact_test.py

```diff
@@ -0,0 +1,52 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+from pathlib import Path
+from fab.artifact import Artifact, New, Unknown
+
+
+class TestArtifact:
+    def test_constructor(self):
+        test_path = Path('/test/path')
+        artifact = Artifact(test_path,
+                            Unknown,
+                            New)
+
+        assert artifact.location == test_path
+        assert artifact.state is New
+        assert artifact.filetype is Unknown
+        assert artifact.depends_on == []
+        assert artifact.defines == []
+        assert artifact._hash is None
+
+    def test_hash(self, tmp_path: Path):
+        test_path = Path(tmp_path / 'test.foo')
+        test_path.write_text("Lorem ipsum dolor sit")
+        expected_hash = 1463158782
+        artifact = Artifact(test_path,
+                            Unknown,
+                            New)
+        assert artifact._hash is None
```

Comment:
Not sure about testing nominally internal private variables. If someone goes accessing them then all bets are off. Therefore what value they contain is not important, or defined.

Reply:
I agree, _in principle_ - though I wanted to try and ensure that the hash hasn't been calculated prior to being requested (one of the problems with it being cached; checking it it is set via the `hash` property will cause it to be calculated!)  I can remove this if you think it's okay without it but can you think of any other way to check the hash part?

Reply:
Thinking about it a little further - perhaps I can do something similar to the other test you commented on below - if I setup the artifact with a non-existent path then I _wouldn't_ expect it to raise whatever type of error it would raise when calling the constructor, but I _would_ expect it to get raised when I try to access the `hash` property (I'm not sure how sensible an idea this is though)

Reply:
I think we now have 3 possible ways forward. I am happy to accept any one of them:

  * Leave it as it is
  * Remove the test on internal variables
  * Test on a none existent file.

Reply:
Let's remove it I think - there may be a better way to test this which we can implement later